Platform/AlderlakeBoardPkg/Library/Stage2BoardInitLib/AcpiPlatform.c

```diff
@@ -893,7 +893,11 @@ PlatformUpdateAcpiGnvs (
   Length = GetPchMaxSerialIoUartControllersNum ();
   for (Index = 0; Index < Length ; Index++) {
     PchNvs->UM0[Index] = FspsConfig->SerialIoUartMode[Index];
-    PchNvs->UC0[Index] = SerialIoUartPciCfgBase(Index);
+    if (FspsConfig->SerialIoUartMode[Index] == 3) { // for SerialIoUartCom only
+      PchNvs->UC0[Index] = (UINT64)(PCH_SERIAL_IO_BASE_ADDRESS + 0x1F000 + (0x2000 * Index));
```

Comment:
@stanleyintel would you help me understand why this config space gets assigned this base address when it's com mode? thanks

Reply:
Thank you for your comment.

When a UART device is configured in COM mode, its PCIe config space is hidden. According to the PCH BIOS Spec, BIOS must use fixed addresses for these BARs so that ACPI entries can use them. Although the fixed address is defined (in PCH BIOS Spec), the offset is assigned by FSP-S. On ADL (ADL-N/ADL-P/ADL-S and RPL), the PCH_SERIAL_IO_BASE_ADDRESS + **0x1F000** + (0x2000 * Index) are used as the alternative access to the PCI configuration space for UART devices.

I also updated the LPSS_UART_TEMP_BASE_ADDRESS for ADL to match FSP-S by using PCH_SERIAL_IO_BASE_ADDRESS + **0x1E000** + (0x2000 * Index) as the default MMIO space